scripts/test_all.py

```diff
@@ -0,0 +1,45 @@
+import os
+import csv
+from argparse import ArgumentParser
+from .preprocess import preprocess
+from .test_single import test_single
+from .load_tools import setup_psiminer, setup_code2seq
+
+
+def test_all(dataset_path: str, model_path: str, results_path: str):
+    """Evaluate and store all"""
+
+    project_names = os.listdir(dataset_path)
+    for project_name in project_names:
+        preprocess(os.path.join(dataset_path, project_name))
+
+    project_names = os.listdir("datasets")
+    result_file = os.path.join(results_path, "results.csv")
+    header = ["Project", "F1", "Precision", "Recall", "Loss"]
+    with open(result_file, "w") as f:
```

Comment:
It is better to use pandas and DataFrames to work with csvs. Most of the time it makes the code shorter and cleaner. However, let’s keep it as it is for now, just use pandas in the future.